### ISSUE TITLE
[FEAT EXTRACTION]

### ISSUE THREAD

**Issue by lcerdaal** [AIM-Harvard/pyradiomics#672]
Hello,

I have extracted GLCM features with and without the voxel-based option offered by pyradiomics. To simplify, I am using a MR image (containing 20 slices) and a mask of the same size, where all slices have 0s except in one of them, which contains the shape of a tumor (in 1s).

I was surprised to see that obtaining the voxel-based map (e.g. for the GLCM entropy), and calculating the mean of the resulting values gives a very different value (4.544) than calculating this same radiomic feature in the traditional way (without the voxel-based option) (9.361). I am using the same parameters in both cases:

imageType:
  Original: {}

featureClass:
  glcm:
    - JointEntropy

setting:
  binWidth: 25
  force2D: true
  label: 1

voxelSetting:
  kernelRadius: 1
  maskedKernel: false
  initValue: nan
  voxelBatch: 10

I have enabled/disabled the maskedKernel option, and the results do not change either. Please, could you help me to understand why this is happening and how to make both results more compatible? (I know they might never be the same, but one would think they should be a bit closer...).

Thank you very much!
Leonor

**Comment by JoostJM** [comment]
TLDR: Entropy calculated segment-based reflects global entropy, calculated voxel-based, then averaged reflects average of local entropy, which can be very different. I.e. not a bug, just different meaning of your feature value.

That's no surprise, and in fact, I would be surprised if it were otherwise. When calculating voxel-based, the values are based on a (fairly small) region surrounding each voxel. That way, you get a local estimate of entropy.
When calculating segment based, the region is much larger, and the feature reflects global entropy. In the local areas the entropy may reflect more homogeneous tissue (i.e. low), while still reflect large entropy over the global tissue. 

consider a very homogenous image, just 4 big squares with differing gray values:
Most voxels will have an entirely flat region, with entropy reflecting this, just at the edge between the squares will the entropy by different. By taking the average of de voxel based map, this will reflect a very homogenous tissue, much more so than the global entropy calculated on the same image.
